cuqi/distribution/_implicit_regularized_gaussian.py

```diff
@@ -0,0 +1,278 @@
+from cuqi.utilities import get_non_default_args
+from cuqi.distribution import Distribution, Gaussian, GMRF
+from cuqi.solver import ProjectNonnegative, ProjectBox, ProximalL1
+
+import numpy as np
+
+
+class ImplicitRegularizedGaussian(Distribution):
+    """ Implicit Regularized Gaussian distribution.
+
+    Defines a Gaussian distribution with implicit regularization. The regularization can be defined
+    in the form of a proximal operator or a projector. Alternatively, preset constraints and regularization
+    can be used.
+
+    Only one of proximal, projector, constraint or regularization can be provided. If none of them are provided,
+    a nonnegativity constraint is used by default.
+
+    Distribution can be used as a prior in a posterior which can be sampled with the RegularizedLinearRTO sampler.
+
+
+    For more details on implicit regularized Gaussian see the following paper:
+
+    [1] Everink, Jasper M., Yiqiu Dong, and Martin S. Andersen. "Sparse Bayesian inference with regularized
+    Gaussian distributions." Inverse Problems 39.11 (2023): 115004.
+
+    Parameters
+    ----------
+    mean
+        See :class:`~cuqi.distribution.Gaussian` for details.
+
+    cov
+        See :class:`~cuqi.distribution.Gaussian` for details.
+
+    prec
+        See :class:`~cuqi.distribution.Gaussian` for details.
+
+    sqrtcov
+        See :class:`~cuqi.distribution.Gaussian` for details.
+
+    sqrtprec
+        See :class:`~cuqi.distribution.Gaussian` for details.
+
+    proximal : callable f(x, scale) or None
+        Euclidean proximal operator of the regularization function g, that is, a solver for the optimization problem
+        min_z 0.5||x-z||_2^2+scale*g(x).

```

Comment:
```suggestion
    proximal : callable f(x, scale) or None
        Euclidean proximal operator of the regularization function f, that is, a solver for the optimization problem
        min_z 0.5||x-z||_2^2+scale*f(x).

```

Reply:
Thanks for the suggestion. I chose not to use f in the optimization problem as it is a different function from the f in ` proximal : callable f(x, scale)`

Reply:
Got it! Thanks for the clarification. I editied my comment (just wanted to emphasize where f comes into play)